src/big/big_float.cr

```diff
@@ -18,7 +18,39 @@ struct BigFloat < Float
   end
 
   def initialize(num : Number)
-    LibGMP.mpf_init_set_d(out @mpf, num.to_f64)
+    # XXX: this case is workaround of Crystal's unrealiable method overloading
+    # remove it when separate BigFloat.new(BigInt) will pass the spec
```

Comment:
This is strange https://github.com/crystal-lang/crystal/pull/4653#issuecomment-313629658 case.

For some reason compiler may prefer not exact overload.